dts/arm/adi/max32/max32690-pinctrl.dtsi

```diff
@@ -0,0 +1,771 @@
+/*
+ * Copyright (c) 2023-2024 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/pinctrl/max32-pinctrl.h>
+
+/ {
+	soc {
+		pinctrl: pin-controller@40008000 {
+
+			/omit-if-no-ref/ spixr_sdio0_p0_1: spixr_sdio0_p0_1 {
+				pinmux = <MAX32_PINMUX(0, 1, AF1)>;
+			};
+
+			/omit-if-no-ref/ spixf_sdio0_p0_1: spixf_sdio0_p0_1 {
+				pinmux = <MAX32_PINMUX(0, 1, AF2)>;
+			};
```

Comment:
suggestion: pinctrl grouping approach is better, this representation (even if used by some vendors) is pretty dumb.

Reply:
Thanks for suggestion, noted, will apply it when get time. for now please process with existing method.